### PR TITLE
uftrace: Add .clang-format file for better code styling

### PR DESCRIPTION
**Pull request by honggyukim**
clang-format describes configurable formatting style options.
It'd be useful when asking someone to follow coding-style guide.

Since uftrace almost follow Linux kernel coding style guide, this patch
copies .clang-format from Linux kenel source and adopt it to follow
uftrace coding style.

Link: https://clang.llvm.org/docs/ClangFormat.html
Link: https://clang.llvm.org/docs/ClangFormatStyleOptions.html
Signed-off-by: Honggyu Kim <honggyu.kp@gmail.com>

### CODE REVIEW

**Comment by honggyukim** [comment]
It can be tested like below.
```
$ clang-format -i cmds/record.c
```
Some of the changes are not satisfy but it can enhanced by modifying `.clang-format` for better styling later on.

**Comment by honggyukim** [comment]
Right. I will take a look at it later. Thanks.

**Comment by honggyukim** [comment]
The following example can be a useful reference.
- https://github.com/git/git/blob/master/.clang-format

File: utils/auto-args.h [hunk 1]
```diff
@@ -252,227 +251,224 @@ static char *auto_args_list =
...
-	"dup@retval/d32;"
-	"dup2@retval/d32;"
-	"sleep@retval/u;"
-	"usleep@retval/d32;"
-	"clock_getres@retval/d32;"
-	"clock_gettime@retval/d32;"
-	"clock_settime@retval/d32;"
-;
+static char *auto_retvals_list = "_Znwm@retval/x;"
+				 "_Znam@retval/x;"
```
Comment (honggyukim) [inline utils/auto-args.h:257]: This auto-generated file shouldn't be touched.

**Review by namhyung — COMMENTED** [review]
I think it's good for contributors to have it installed in their local system.  Could you please add it to CONTRIBUTING.md (and replace coding style part)?

**Comment by honggyukim** [comment]
Sure. I will update the doc as well.

**Comment by honggyukim** [comment]
> I think it's good for contributors to have it installed in their local system. Could you please add it to CONTRIBUTING.md (and replace coding style part)?

I've just updated the document as follows.
https://github.com/namhyung/uftrace/blob/1f3ffcc1ec9baed71ece4ac370cc1cbb650a726a/CONTRIBUTING.md

**Comment by honggyukim** [comment]
The unit test result is fine.
```
unit test stats
====================
 83 ran successfully
  0 failed
  0 skipped
  0 signal caught
  0 unknown result
```
And the system result is as follows.
```
$ cat failed-tests.txt
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
117 time_range          : OK OK OK OK OK OK OK NG OK OK
125 report_range        : OK OK OK OK OK OK OK OK OK NG
151 recv_runcmd         : NG NG NG NG NG NG NG NG NG NG
162 pltbind_now_pie     : BI BI BI BI BI BI BI BI BI BI
182 thread_exit         : OK OK OK OK OK OK NG OK NG NG
202 arg_dwarf2          : OK OK NG NG NG SK SK SK SK SK
203 arg_dwarf3          : OK OK NG NG NG SK SK SK SK SK
204 arg_dwarf4          : OK OK NG NG OK SK SK SK SK SK
208 watch_cpu           : OK OK OK OK OK OK OK OK OK NG
238 report_field2       : NZ NZ NZ NZ NZ NZ NZ NZ NZ NZ
251 exception4          : NG NG NG NG NG NG NG NG NG NG
```
I think they are all known issues so I can confirm that the clang-format does not break our tests.

**Review by namhyung — APPROVED** [review]
LGTM

**Pull request merged**